runtime/databricks/automl_runtime/forecast/utils.py

```diff
@@ -187,6 +187,33 @@ def generate_cutoffs(df: pd.DataFrame, horizon: int, unit: str,
         )
     return list(reversed(result))
 
+def generate_custom_cutoffs(df: pd.DataFrame, horizon: int, unit: str,
+                     split_cutoff: pd.Timestamp) -> List[pd.Timestamp]:
+    """
+    Generate custom cutoff times for cross validation based on user-specified split cutoff.
+    Period (step size) is 1.
+    :param df: pd.DataFrame of the historical data.
+    :param horizon: int number of time into the future for forecasting.
+    :param unit: frequency unit of the time series, which must be a pandas offset alias.
+    :param split_cutoff: the user-specified cutoff, as the starting point of cutoffs
+    :return: list of pd.Timestamp cutoffs for cross-validation.
+    """
+    period = 1 
+    period_dateoffset = pd.DateOffset(**DATE_OFFSET_KEYWORD_MAP[unit])*period
+    horizon_dateoffset = pd.DateOffset(**DATE_OFFSET_KEYWORD_MAP[unit])*horizon
+
+    cutoff = split_cutoff
+    result = [cutoff]
+    while result[-1] <= max(df["ds"]) - horizon_dateoffset:
```

Comment:
nit: if `df` and `horizon_dataoffset` are not updated inside the while loop, maybe move it before the loop to be more efficient.

Reply:
Fixed.